chapters/11-version-control-workflow.Rmd

```diff
@@ -1,18 +1,244 @@
+    Articles that are ignored from package build. (we discussed this a bit
+    already, needs more discussion)
+- Branching/forks
+    - Text: what is it, power of it, features.
+- Pull Requests
+    - Exercises: Reading, comprehension, and conceptual understanding (non-coding)
+- Security:
+    - Text: why its important, how to do it.
+    - Code:
+        - usethis::git_sitrep()
+        - usethis::gh_token_help()
+        - usethis::create_github_token()
+    - Text: Save PAT in password manager (they should have this anyway, but this forces them)
+    - Code: 
+        - gitcreds::gitcreds_set(), everytime computer restarts, only saved for session
+    - Exercises: Conceptual understanding of PAT, tokens, and password managers (non-coding)
+    - Exercises: Setting up PAT, saving PAT, using gitcreds at start of session
+- usethis pr_* helpers
+    - Text: Why use these over GitHub interface
+- As contributor/sole developer/as team
+    - usethis::pr_init()
+        - Make changes to Contributing about using pr helpers
+    - usethis::pr_push()
+    - usethis::pr_view()
+    - usethis::pr_pause()
+    - Exercise: Make an edit on the PR changed files using GitHubs edit interface.
+    - Exercise: Create another PR for adding ...FIXME.
+    - usethis::pr_resume()
+    - usethis::pr_pull()
+    - usethis::pr_merge_main()
+    - usethis::pr_push()
+    - PR has been merged
+    - gert::git_branch_list() # Show list of created branches
+    - usethis::pr_finish()
+    - Exercise: Do the same process for the other PR you had created.
+    - gert::git_branch_list() # Show list of created branches
+    - usethis::pr_init()
+    - usethis::pr_forget()
+    - gert::git_branch_list() # Show list of created branches
+- As maintainer, reviewer, team lead. (Hard to do exercises for this... need to
+think how this part would be implemented/practiced.)
+    - usethis::pr_view()
+    - usethis::pr_fetch()
+    - usethis::pr_push()
+    - usethis::pr_merge_main()
+    - usethis::pr_view()
+    - Manually merge PR.
+    - usethis::pr_finish()
+    - Exercise: Conceptual, write out/describe how this process could improve
+    collaboration and teamwork.
+
+## Learning objectives {#11-learning-objectives}
+
+- Learn what branching in Git is, about the concept behind it, and why it's so
+powerful as a software development tool.
+- Learn how to use branching in Git, as well as the related concept of forking
+and Pull Requests in GitHub, by making use of usethis' Pull Request helper
+functions.
+- Understand how the concepts and workflows shown could be applied in team
+or contributor settings and why they are powerful tools for collaboration and
+teamwork.
+
+## Collaborating with Git
+
+### Exercise: Consider ways of collaborating and their effectiveness.
+
+TODO: Taken from PyRSE, needs modification.
+
+Chartreuse and Fuchsia are collaborating on a project.
+Describe what is in each of the four repositories involved after each of the steps below.
```

Comment:
I know this is taken from PyRSE, but can we replace `master` with `main`? This also needs to be reviewed at the book level since I don't believe we have been consistent up to now.

Reply:
Yeesssss, definitely, 100% for sure!